tools/clang/lib/SPIRV/PreciseVisitor.cpp

```diff
@@ -233,14 +233,10 @@ bool PreciseVisitor::visit(SpirvUnaryOp *inst) {
   return true;
 }
 
-bool PreciseVisitor::visit(SpirvNonUniformBinaryOp *inst) {
-  inst->getArg1()->setPrecise(inst->isPrecise());
-  inst->getArg2()->setPrecise(inst->isPrecise());
-  return true;
-}
-
-bool PreciseVisitor::visit(SpirvNonUniformUnaryOp *inst) {
-  inst->getArg()->setPrecise(inst->isPrecise());
+bool PreciseVisitor::visit(SpirvGroupNonUniformOp *inst) {
+  if (inst->isPrecise())
```

Comment:
Comparing to the old code, shouldn't this function always call `setPrecise` on each operand, and set the value to `inst->isPrecise()`? In this case, if `inst->isPrecise()` returns false, and any of operands current have precise set to `true`, they will not be updated. In other words, shouldn't this be:

```
  for (auto *operand : inst->getOperands())
    operand->setPrecise(inst->isPrecise());
  return true;
```

Reply:
It's a default-false flag that get propagated down by this visitor when it's true, so either should be equivalent but switched just to be closer to the original.